src/evict/evict_page.c

```diff
@@ -530,12 +521,14 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
         WT_RET(ret);
     }
 
-    /*
-     * It is always OK to evict pages from dead trees if they don't have children.
-     */
+    /* It is always OK to evict pages from dead trees if they don't have children. */
     if (F_ISSET(session->dhandle, WT_DHANDLE_DEAD))
         return (0);
 
+    /* Don't attempt to evict fast-truncate pages until the truncate completes. */
```

Comment:
> Doesn't this check is already verified as part of `__wt_page_can_evict`?

A good question, but yes, even with the check in `__wt_page_can_evict()`, you can still get into here with `ref->ft_info.update != NULL`, I've changed that line into an assert, and it will fire with enough cache pressure.

It seems to me eviction won't call `__wt_page_can_evict()` until `WT_REF.page` is set, that is, until the fast-truncate page is instantiated, which means that if all eviction paths call `__wt_page_can_evict()`, it shouldn't happen.

I've never tracked down the case where it can happen, if someone wants to state as a fact that `__wt_page_can_evict()` is called on all eviction paths before queueing a page for eviction, then I should dig further.

@agorrod?

Reply:
The `__wt_page_can_evict()` call happens 10 lines below in a condition, but that will be always true in a normal run.

Reply:
Thank you @kommiharibabu, I see what you're saying, there should be no reason for my additional test. Thanks!